### PR TITLE
Feat/phone text field

### PR DESCRIPTION
**Pull request by NaamanTsur**
* Created numeric text field to only allow number typing.
* Changed all phone text fields to use numeric text fields.

### CODE REVIEW

**Review by ItayBenMoshe — CHANGES_REQUESTED** [review]
There is another phone text field in ContactForm, also change id to be numeric as well

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/BusinessContactForm/BusinessContactForm.tsx [hunk 1]
```diff
@@ -71,11 +72,14 @@ const BusinessContactForm: React.FC = (): JSX.Element => {
                             name={InteractionEventDialogFields.CONTACT_PERSON_PHONE_NUMBER}
                             control={control}
                             render={(props) => (
-                                <TextField
+                                <NumericTextField
+                                    name={props.name}
                                     value={props.value}
-                                    onChange={event => props.onChange(event.target.value as string)}
-                                    error={get(errors, props.name)}
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/BusinessContactForm/BusinessContactForm.tsx:77]: why did you delete error attribute ?

File: client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/BusinessContactForm/BusinessContactForm.tsx [hunk 2]
```diff
@@ -71,11 +72,14 @@ const BusinessContactForm: React.FC = (): JSX.Element => {
...
-                                    label={get(errors, props.name)?.message}
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/InteractionsTab/InteractionEventForm/BusinessContactForm/BusinessContactForm.tsx:78]: also why?

File: client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTab.tsx [hunk 3]
```diff
@@ -194,17 +195,18 @@ const PersonalInfoTab: React.FC<Props> = ({ id, onSubmit }: Props): JSX.Element
...
                                     className={classes.phoneInput}
                                     value={props.value}
-                                    onChange={(newValue) => (
-                                        props.onChange(newValue.target.value)
+                                    onChange={(newValue: string) => (
+                                        props.onChange(newValue)
                                     )}
                                     placeholder={PHONE_LABEL}
-                                    error={errors[PersonalInfoDataContextFields.PHONE_NUMBER]}
-                                    label={errors[PersonalInfoDataContextFields.PHONE_NUMBER] ? errors[PersonalInfoDataContextFields.PHONE_NUMBER]?.message : 'טלפון*'}
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTab.tsx:207]: I think you should keep 'טלפון*' label

File: client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTab.tsx [hunk 4]
```diff
@@ -224,18 +226,19 @@ const PersonalInfoTab: React.FC<Props> = ({ id, onSubmit }: Props): JSX.Element
...
                                     className={classes.phoneInput}
                                     value={props.value}
-                                    onChange={(newValue) => (
-                                        props.onChange(newValue.target.value)
+                                    onChange={(newValue: string) => (
+                                        props.onChange(newValue)
                                     )}
                                     placeholder={PHONE_LABEL}
-                                    error={errors[PersonalInfoDataContextFields.ADDITIONAL_PHONE_NUMBER]}
-                                    label={errors[PersonalInfoDataContextFields.ADDITIONAL_PHONE_NUMBER] ? errors[PersonalInfoDataContextFields.ADDITIONAL_PHONE_NUMBER]?.message : 'טלפון'}
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTab.tsx:237]: same as above

File: client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTab.tsx [hunk 5]
```diff
@@ -254,18 +257,19 @@ const PersonalInfoTab: React.FC<Props> = ({ id, onSubmit }: Props): JSX.Element
...
                                     className={classes.phoneInput}
                                     value={props.value}
-                                    onChange={(newValue) => (
-                                        props.onChange(newValue.target.value)
+                                    onChange={(newValue: string) => (
+                                        props.onChange(newValue)
                                     )}
                                     placeholder={PHONE_LABEL}
-                                    error={errors[PersonalInfoDataContextFields.CONTACT_PHONE_NUMBER]}
-                                    label={errors[PersonalInfoDataContextFields.CONTACT_PHONE_NUMBER] ? errors[PersonalInfoDataContextFields.CONTACT_PHONE_NUMBER]?.message : 'טלפון'}
```
Comment (ItayBenMoshe) [inline client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTab.tsx:267]: same

File: client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTab.tsx (hunk 3 above)
Comment (NaamanTsur) [reply client/src/components/App/Content/InvestigationForm/TabManagement/PersonalInfoTab/PersonalInfoTab.tsx:207]: I brought them all back. We don't need the error logic in here anymore so I only provide the normal label.

**Review by ItayBenMoshe — APPROVED** [review]

**Review by yaniryos — APPROVED** [review]
